daffodil-tdml-processor/src/main/scala/org/apache/daffodil/tdml/processor/DaffodilTDMLDFDLProcessor.scala

```diff
@@ -139,18 +139,31 @@ final class TDMLDFDLProcessorFactory private (
     }
   }
 
+
   private def compileProcessor(
     schemaSource: DaffodilSchemaSource,
     useSerializedProcessor: Boolean,
     optRootName: Option[String],
     optRootNamespace: Option[String]): TDML.CompileResult = {
-    val pf = compiler.compileSource(schemaSource, optRootName, optRootNamespace)
-    val diags = pf.getDiagnostics
-    if (pf.isError) {
-      Left(diags)
+    //
+    // if the schemaSource is not an XSD file, then we assume it is
+    // a pre-compiled DFDL schema
+    //
+    if (schemaSource.isXSD)
+    {
+      val pf = compiler.compileSource(schemaSource, optRootName, optRootNamespace)
+      val diags = pf.getDiagnostics
+      if (pf.isError) {
+        Left(diags)
+      } else {
+        val res = this.generateProcessor(pf, useSerializedProcessor)
+        res
+      }
     } else {
-      val res = this.generateProcessor(pf, useSerializedProcessor)
-      res
+      val chan = Channels.newChannel(schemaSource.uriForLoading.toURL.openStream())
+      val dp = compiler.reload(chan)
+      val diags = dp.getDiagnostics
+      Right((diags, new DaffodilTDMLDFDLProcessor(dp)))
```

Comment:
In general, I think it's a not a great idea to store precompiled binaries in a repository along with the schema/tests. Any update to Daffodil (including snapshots) are likely to not be compatible, so it's going to be pretty fragile. And any updates to the schema requires you to rebuild the precompiled binary--keeping those in sync sounds difficult and easy to forget.

Ideally we would add a better caching mechanism than what we have right now (e.g. cache to a file instead of to memory). Or as I think you've suggested in the past, when we build the schema jar we also build a precompiled binary into the jar that the TDML runner could look for. That way we can rely on sbt to track when the schema changes and rebuilds the jar/precompiled binary as needed and you don't have to worry about storing one in the repo. That probably requires a special Daffodil plugin though or special build.sbt logic, so probably not a trivial amount of work.

This feels fine as a temporary solution, but I hope we are able to deprecated it relatively soon and not make this standard practice.

Reply:
Completely agree that they shouldn't be stored in repos. My use case for this keeps them in /tmp. 

The only reason there is this fake ".bin" in here is for purposes of having a test here that exercises the code paths.

re: improving this feature further... I did think about that, but punted for now. It's not THAT hard, there's just lots of time pressure now on other things. But I definitely wanted to put this into Daffodil while we can rather than write yet another non-TDML test rig to dance around this problem. 

We could add the method to daffodil to allow one to get latest mod date for any file included/imported by a schema, and use that date to decide whether to recompile or not. Having daffodil do this avoids reinventing include/import and resolver logic again elsewhere. I think this is pretty easy. DFDLSchemaFile class gets a modDateTime member, and rollups doing max on that for an entire SchemaSet. 

Still doesn't deal with what if the schema is the same, but you changed daffodil itself, but it would still be better than entirely manual.